iquip/apps/scheduler.py

```diff
@@ -281,55 +292,60 @@ class SchedulerApp(qiwis.BaseApp):
     """
 
     def __init__(self, name: str, parent: Optional[QObject] = None):
-        """Extended."""  
+        """Extended.
+
+        Args:
+            name: The name of the app.
+            parent: The QObject that contains this app."""
         super().__init__(name, parent=parent)
         self.schedulerFrame = SchedulerFrame()
         # TODO(giwon2004): Below are for testing before connecting to artiq-proxy.
-        self.addExperiment(ExperimentInfo("exp1", {"rid": 9, "priority": 3}))
-        self.addExperiment(ExperimentInfo("exp2", {"rid": 10, "priority": 3}))
-        self.addExperiment(ExperimentInfo("exp3", {"rid": 11, "priority": 4}))
-        self.runExperiment(ExperimentInfo("exp4", {"rid": 12, "priority": 2}))
 
     # TODO(giwon2004): Below are called by the signal from artiq-proxy.
-    def runExperiment(self, info: Optional[ExperimentInfo] = None):
+    def runExperiment(self, exp: Optional[ExperimentView] = None):
         """Sets the experiment onto 'currently running' section.
 
         Args:
-            info: The experiment information.
+            exp: The experimentView instance to be run. None if there is no experiements running.
         """
-        self.schedulerFrame.expRun.updateInfo(info)
+        if exp is not None:
+            self.schedulerFrame.runningView.updateInfo(exp.experimentInfo)
+        else:
+            self.schedulerFrame.runningView.updateInfo(None)
+        if exp in self.schedulerFrame.model.experimentData:
+            self.deleteExperiment(exp)
 
-    def addExperiment(self, info: ExperimentInfo):
-        """Adds the experiment to 'queued experiments' section (expList).
+    def addExperiment(self, exp: ExperimentView):
+        """Adds the experiment to 'queued experiments' section (queueView).
 
         Args:
-            info: The experiment information.
+            exp: The experimentView instance to be added.
         """
-        self.schedulerFrame.model.experimentData.append(info)
-        self.schedulerFrame.model.experimentData.sort(key=lambda x: x.arginfo["priority"],
+        self.schedulerFrame.model.experimentData.append(exp)
+        self.schedulerFrame.model.experimentData.sort(key=lambda x: x.priority(),
```

Comment:
FYI (not mandatory): [`operator.methodcaller()`](https://docs.python.org/3/library/operator.html#operator.methodcaller)
E.g.: `....sort(key=operator.methodcaller("priority"), reverse=True)`